xlators/protocol/server/src/server-rpc-fops_v2.c

```diff
@@ -3186,19 +3215,44 @@ int
 server4_lookup_resume(call_frame_t *frame, xlator_t *bound_xl)
 {
     server_state_t *state = NULL;
+    dict_t *xdata = NULL;
 
     state = CALL_STATE(frame);
 
     if (state->resolve.op_ret != 0)
         goto err;
 
-    if (!state->loc.inode)
+    xdata = state->xdata ? dict_ref(state->xdata) : dict_new();
```

Comment:
I think we shouldn't allow a failure of dict_new(). Otherwise we risk missing a namespace.

Reply:
ack. Treating it as error now.